### PR TITLE
refactor(client/android): remove dependency on serverName in the tunnel config  

### PR DESCRIPTION
**Pull request by 62w71st**
I tested the change by looking at the logs: 

2024-08-04 21:26:58.088 11282-11340 OutlinePlugin           org.outline.android.client           I  Starting VPN tunnel 3cdcb995-2f4f-4102-bfee-23aa89ac18b5 for server United Kingdom #1097 / OutlineKeys.com
2024-08-04 21:26:58.206 11343-11354 VpnTunnelService        org.outline.android.client           I  Starting tunnel 3cdcb995-2f4f-4102-bfee-23aa89ac18b5 for server United Kingdom #1097 / OutlineKeys.com

Some notes on foreground services:
1) our manifest does not define them, so that's probably why we don't see them. I tested that we don't get foreground notifications in play store's Outline either: https://developer.android.com/develop/background-work/services/foreground-services#types
2) In Android 14, setting ongoing flag on notifications no longer makes them non-dismissable: https://developer.android.com/about/versions/14/behavior-changes-all#non-dismissable-notifications

### CODE REVIEW

**Comment by 62w71st** [comment]
Just signed CLA

File: client/src/www/app/outline_server_repository/server.cordova.ts [hunk 1]
```diff
@@ -23,8 +23,12 @@ export class CordovaTunnel implements PlatformTunnel {
     if (!config) {
       throw new errors.IllegalServerConfiguration();
     }
-    return pluginExecWithErrorCode<void>('start', this.id, name, config);
-  }
+    if (cordova.platformId === 'android') {
+      return pluginExecWithErrorCode<void>('start', this.id, name, config);
+    } else {
+      // TODO: implement serverName in iOS.
+      return pluginExecWithErrorCode<void>('start', this.id, config);
```
Comment (sbruens) [inline client/src/www/app/outline_server_repository/server.cordova.ts:30]: Please merge #2075, then you can get rid of this TODO as the basic wiring was done by @fortuna in  https://github.com/Jigsaw-Code/outline-apps/pull/2075/files#diff-509f058f9661f2600b457b0d73100f9e57d98e705b7b192b63a20cebda53df6d.

**Review by sbruens — CHANGES_REQUESTED** [review]
Nice, thanks! Please merge master as per my comment.

File: client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java [hunk 2]
```diff
@@ -474,17 +475,18 @@ private void startLastSuccessfulTunnel() {
     try {
       final String tunnelId = tunnel.getString(TUNNEL_ID_KEY);
```
Comment (fortuna) [inline client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java:476]: The three keys are a bit useless and requires more wiring. Can we have a single TunnelConfig with a format like:

- id: string
- name: string
- transport: TransportConfig (the current config, should become an opaque string)

?

It simplifies the code and we will need that container for VPN-level settings, like DNS.

Comment (62w71st) [reply client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java:476]: IIUC the proposal is  to change the TunnelConfig struct here: https://github.com/Jigsaw-Code/outline-apps/blob/6e78697c7bea2d6aa7be7dc5b1223bc12fc665fb/client/src/cordova/android/OutlineAndroidLib/outline/src/main/aidl/org/outline/TunnelConfig.aidl#L19

That's do-able but it's a bigger change. Can that be done incrementally in a different PR or does it have to be in this PR?

File: client/src/www/app/outline_server_repository/server.cordova.ts (hunk 1 above)
Comment (62w71st) [reply client/src/www/app/outline_server_repository/server.cordova.ts:30]: Thanks! will do.

File: client/src/www/app/outline_server_repository/server.cordova.ts [hunk 3]
```diff
@@ -23,8 +23,12 @@ export class CordovaTunnel implements PlatformTunnel {
     if (!config) {
       throw new errors.IllegalServerConfiguration();
     }
-    return pluginExecWithErrorCode<void>('start', this.id, name, config);
-  }
+    if (cordova.platformId === 'android') {
```
Comment (fortuna) [inline client/src/www/app/outline_server_repository/server.cordova.ts:26]: Please revert this file and pull master. The wiring is already implemented for all platforms in https://github.com/Jigsaw-Code/outline-apps/pull/2105.

File: client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java (hunk 2 above)
Comment (fortuna) [reply client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java:476]: Sounds good.

**Review by fortuna — APPROVED** [review]

**Comment by 62w71st** [comment]
To get around the CLA issue, I squashed commits into one and made another PR at: https://github.com/Jigsaw-Code/outline-apps/pull/2112

**Pull request closed without merging**